presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestMaterializedViewCandidateExtractor.java

```diff
@@ -0,0 +1,215 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.analyzer;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.SystemSessionProperties;
+import com.facebook.presto.common.QualifiedObjectName;
+import com.facebook.presto.metadata.SessionPropertyManager;
+import com.facebook.presto.sql.parser.SqlParser;
+import com.facebook.presto.sql.tree.Query;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.facebook.presto.testing.assertions.Assert.assertEquals;
+import static java.lang.String.format;
+
+public class TestMaterializedViewCandidateExtractor
+{
+    private static Session session = testSessionBuilder(new SessionPropertyManager(new SystemSessionProperties())).build();
+    private static SqlParser sqlParser = new SqlParser();
+
+    @Test
+    public void testWithSimpleQuery()
+    {
+        String baseTable = "base_table";
+        QualifiedObjectName baseTableName = QualifiedObjectName.valueOf("catalog.schema.base_table");
+        QualifiedObjectName materializedViewName = QualifiedObjectName.valueOf("catalog.schema.view");
+
+        Map<QualifiedObjectName, List<QualifiedObjectName>> baseTableToMaterializedViews = new HashMap<>();
+        baseTableToMaterializedViews.put(baseTableName, ImmutableList.of(materializedViewName));
+
+        Set<QualifiedObjectName> expectedMaterializedViewCandidates = new HashSet<>();
+        expectedMaterializedViewCandidates.add(materializedViewName);
+
+        String baseQuerySql = format("SELECT x, y From %s", baseTable);
+
+        assertCandidateMaterializedView(expectedMaterializedViewCandidates, baseQuerySql, baseTableToMaterializedViews);
+    }
+
+    @Test
+    public void testWithAliasedRelation()
+    {
+        String baseTable = "base_table";
+        QualifiedObjectName baseTableName = QualifiedObjectName.valueOf("catalog.schema.base_table");
+        QualifiedObjectName materializedViewName = QualifiedObjectName.valueOf("catalog.schema.view");
+
+        Map<QualifiedObjectName, List<QualifiedObjectName>> baseTableToMaterializedViews = new HashMap<>();
+        baseTableToMaterializedViews.put(baseTableName, ImmutableList.of(materializedViewName));
+
+        Set<QualifiedObjectName> expectedMaterializedViewCandidates = new HashSet<>();
```

Comment:
may be replace this with
Set<QualifiedObjectName> expectedMaterializedViewCandidates = ImmutableSet.of(materializedViewName);